packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +290,102 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed']
```

Comment:
This logic has changed for Foreman 2.0 to also take the provider into account (https://github.com/theforeman/puppet-foreman_proxy/commit/54067fe5ac742000d3c2cdb8505d8f191383aeb7)

Reply:
thanks, updating